lhotse/features/io.py

```diff
@@ -56,6 +58,62 @@ def storage_path(self) -> str:
     def write(self, key: str, value: np.ndarray) -> str:
         ...
 
+    def store_array(
+        self,
+        key: str,
+        value: np.ndarray,
+        frame_shift: Optional[Seconds] = None,
+        temporal_dim: Optional[int] = None,
+        start: Seconds = 0,
+    ) -> Union[Array, TemporalArray]:
+        """
+        Store a numpy array in the underlying storage and return a manifest
+        describing how to retrieve the data.
+
+        If the array contains a temporal dimension (e.g. it represents the
+        frame-level features, alignment, posteriors, etc. of an utterance)
+        then ``temporal_dim`` and ``frame_shift`` may be specified to enable
+        downstream padding, truncating, and partial reads of the array.
+
+        :param key: An ID that uniquely identifies the array.
+        :param value: The array to be stored.
+        :param frame_shift: Optional float, when the array has a temporal dimension
+            it indicates how much time has passed between the starts of consecutive frames
+            (expressed in seconds).
+        :param temporal_dim: Optional int, when the array has a temporal dimension,
+            it indicates which dim to interpret as temporal.
+        :param start: Float, when the array is temporal, it indicates what is the offset
+            of the array w.r.t. the start of recording. Useful for reading subsets
+            of an array when it represents something computed from long recordings.
+            Ignored for non-temporal arrays.
+        :return: A manifest of type :class:`~lhotse.array.Array` or
+            :class:`~lhotse.array.TemporalArray`, depending on the input arguments.
+        """
+        is_temporal = frame_shift is not None and temporal_dim is not None
+        if not is_temporal:
+            assert all(arg is None for arg in [frame_shift, temporal_dim]), (
+                "frame_shift and temporal_dim have to be both None or both set "
+                f"(got frame_shift={frame_shift}, temporal_dim={temporal_dim})."
+            )
+
+        storage_key = self.write(key, value)
+        array = Array(
+            storage_type=self.name,
+            storage_path=self.storage_path,
```

Comment:
Note: This is problematic.

`self.storage_path` is not a `str` and hence it is not serializable.

Should be
```
storage_path=str(self.storage_path),
```

---

The  following is the exception that I get without the above change
```
  File "./conformer_ctc/ali.py", line 362, in main
    cut_set.to_json(args.out_cut_manifest)
  File "/ceph-fj/fangjun/open-source-2/lhotse-ali/lhotse/serialization.py", line 67, in to_json
    save_to_json(list(self.to_dicts()), path)
  File "/ceph-fj/fangjun/open-source-2/lhotse-ali/lhotse/serialization.py", line 55, in save_to_json
    json.dump(data, f, indent=2)
  File "/root/fangjun/open-source/pyenv/versions/3.8.6/lib/python3.8/json/__init__.py", line 179, in dump
    for chunk in iterable:
  File "/root/fangjun/open-source/pyenv/versions/3.8.6/lib/python3.8/json/encoder.py", line 429, in _iterencode
    yield from _iterencode_list(o, _current_indent_level)
  File "/root/fangjun/open-source/pyenv/versions/3.8.6/lib/python3.8/json/encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "/root/fangjun/open-source/pyenv/versions/3.8.6/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/root/fangjun/open-source/pyenv/versions/3.8.6/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/root/fangjun/open-source/pyenv/versions/3.8.6/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  [Previous line repeated 1 more time]
  File "/root/fangjun/open-source/pyenv/versions/3.8.6/lib/python3.8/json/encoder.py", line 438, in _iterencode
    o = _default(o)
  File "/root/fangjun/open-source/pyenv/versions/3.8.6/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type PosixPath is not JSON serializable
```

Reply:
Good catch